src/main/kotlin/org/codefreak/codefreak/entity/EvaluationStep.kt

```diff
@@ -32,6 +33,12 @@ class EvaluationStep(
   @OneToMany(mappedBy = "evaluationStep", fetch = FetchType.LAZY, cascade = [CascadeType.ALL])
   var feedback = mutableSetOf<Feedback>()
 
+  @OneToOne(mappedBy = "evaluationStep", cascade = [CascadeType.ALL])
+  var pointsOfEvaluationStep: PointsOfEvaluationStep? = null
+
+  @ManyToOne
+  var gradeDefinition: GradeDefinition? = null
```

Comment:
It belongs to so many. :)
Is it an general design issue to keep the GradeDefinition foreign key in multiply Entities? I see why but i find long cascading harder to read. Thats why i keep the GradeDefinition in this case close.

Reply:
I mean "belong" in the sense of ownership. The relations of the ER model have semantics like "has a", "consists of" (aggregation, composition). One needs to have these relations in mind in order to make sense of the model. If we add addtional links on a technical level that don't have the same semantics, it is very confusing. We need to think about ownership and encapsulation. For you, it may be simpler to read if it is more consise but for everyone else it is harder to comprehend.